### DISCUSSION TITLE
/getting-likes-and-comments-from-modern-pages/

### DISCUSSION THREAD

**Discussion by comment-migrator** [estruyf/eliostruyf-comments#291] [Comments]
Getting likes and comments from modern pages
        
[https://www.eliostruyf.com/getting-likes-and-comments-from-modern-pages/](https://www.eliostruyf.com/getting-likes-and-comments-from-modern-pages/)

**Comment by comment-migrator** [comment]
```header
Originally posted by Nigel Price on 2018-03-27 14:37
```

What ! Still no way of knowing when a comment is made on a modern page ? Better turn comments off until Microsoft fix this one, unless you want blasphemous or rude comments on your sites as there is no way of knowing when they appear.

**Reply by comment-migrator** [reply]
```header
Originally posted on 2018-03-27 19:48
```

You can try and bind yourself a flow to the comments list on the root site collection. Probably not supported, but you can give it a shot.

**Reply by comment-migrator** [reply]
```header
Originally posted by Nigel Price on 2018-03-27 20:35
```

Hi Elio - The list disappeared just before Christmas. Comments are now stored in "Cloud Storage" so there is nothing to hang a flow onto. Added to uservoice -> [https://sharepoint.uservoic...](https://sharepoint.uservoice.com/forums/329214-sites-and-collaboration/suggestions/20359924-comment-notifications "https://sharepoint.uservoice.com/forums/329214-sites-and-collaboration/suggestions/20359924-comment-notifications")

**Comment by comment-migrator** [comment]
```header
Originally posted by MITUL RANA on 2019-01-28 10:17
```

How can we get these likes counts and comments via search API, we need this to build some app as this only gives for a perticular list id and in one search results mutiple list ids are their

Like :

"" /\_api/search/query?querytext='IsDocument:TrueAND FileExtension:aspxANDPromotedState:2'&selectproperties='Title,Author,PictureThumbnailURL,SiteTitle,Description,SPWebUrl,OriginalPath,LastModifiedTime,ViewsLifeTime,ViewsRecent,LikesCount,siteId,ListItemID,CusotmComment,CusotmLike,LikeCountLifetime,CommentsCount'&StartRow=0&RowLimit=30""

**Reply by comment-migrator** [reply]
```header
Originally posted on 2019-02-19 15:22
```

If you want to get the likes and comments via the search API, you will first have to map the crawled properties to a managed property. Once that is done, you will see the numbers via search.

Be aware, the numbers will not automatically update with search when likes and counts are added. The numbers will only be updated when the library gets reindexed. So the only way to know the right number is to directly call the likes and comments API.

**Comment by comment-migrator** [comment]
```header
Originally posted by Carlos Alejandro on 2019-04-02 08:09
```

Would this work for Sharepoint List items as well? I would be interested  
in implementing comments for list items but I'm not sure if I should  
look at this API or just add a "append content" multiline column (I  
can't find much info about how to handle this kind of column with  
spfx/rest API)

**Reply by brianpmccullough** [reply]
Did anyone ever figure out if this was possible on other lists/libraries outside of Site Pages?  Or is this "special" functionality within Site Pages?

**Reply by estruyf** [reply]
I checked this a couple of weeks ago. The likes and comments information is available on any list/library, but unfortunately, you can only use the functionality on the Site Pages library.

Here you can see that the likes information is available on a document:

<img width="1271" alt="Likes information on a document library" src="https://github.com/estruyf/eliostruyf-comments/assets/2900833/2849a843-bb42-4b11-9922-670db6190b36">

Trying to like the document will give you an error:

![Like a document will not work](https://github.com/estruyf/eliostruyf-comments/assets/2900833/572c7400-21fe-4659-9abe-39fd4ae95a41)

**Comment by comment-migrator** [comment]
```header
Originally posted by Nitin Logan on 2019-04-10 20:08
```

Thanks for article.

I have few questions:  
1\. Which software are you using to call an API.  
2\. Is there any way to get "\_AuthorByline" column value from Site Pages library.  
3\. How do I get the page view also (Can you let me know if there is any API exist)

Thanks  
Varun

**Reply by comment-migrator** [reply]
```header
Originally posted on 2019-04-18 08:56
```

Thanks

1\. I use my own tool that I've written for this: [https://github.com/estruyf/...](https://github.com/estruyf/spfx-rest-api-tester)  
2\. You can retrieve the Author its user ID from the site: [https://tenant.sharepoint.c...](https://tenant.sharepoint.com/sites/coffee-point/_api/web/Lists/GetByTitle('Site) Pages')/Items()?$select=\_AuthorBylineStringId  
![https://uploads.disquscdn.c...](https://uploads.disquscdn.com/images/f10462f30c8be2fbad4007a431377870863544fd9ccb3668139d8fcdf7e30563.png)

Or you can also get it with the RenderListDataAsStream API. You could use the POST data from above, only thing to add is the following field reference:

![https://uploads.disquscdn.c...](https://uploads.disquscdn.com/images/b9b9a6990587000443e4a8eee0d9cdd8908ceb70c91f229b8c6d8e55fe54a3dc.png)

3\. The page views can be retrieved from search. There are various managed properties that give you more information about this like: ViewsRecent, ViewsLast1Day, ViewsLifeTime, ...

**Reply by comment-migrator** [reply]
```header
Originally posted by Leif Frederiksen on 2020-05-28 13:07
```

Just downloaded your API tester. That's pretty cool - thanks a lot for sharing :-)

**Comment by comment-migrator** [comment]
```header
Originally posted by Francois Souyri on 2019-09-05 14:53
```

Hi @eliostruyf:disqus, have you tried to edit the comment's name author to another by any chance?  
Like impersonify a comment from the president of the company?  
thanks

**Comment by comment-migrator** [comment]
```header
Originally posted by Ricardo Carneiro on 2020-01-16 14:06
```

Hi @eliostruyf:disqus,

Is it possible to get these information on SharePoint 2019 Server?

Thanks.  
Ricardo

**Comment by comment-migrator** [comment]
```header
Originally posted by ATI-Eric Willman on 2020-02-26 19:11
```

There does not appear to be a crawled property to map to for "LikedBy", how can we query for all pages "Liked" by the current user?

**Comment by comment-migrator** [comment]
```header
Originally posted by Kalpesh Vaghela on 2020-04-13 07:33
```

@eliostruyf:disqus Is there any API available to add/set like information in comments? e.g. I want to like particular comment/reply using some user from background code.

**Comment by comment-migrator** [comment]
```header
Originally posted by Ruchi on 2020-10-22 15:45
```

There is also an option to turn off the comments in a SharePoint page.  
Is it possible to check whether the comments is turned off or not using REST?

**Comment by comment-migrator** [comment]
```header
Originally posted by Shabana Mohammed on 2021-10-07 09:32
```

I am able to retrieve the Likes and Comments with the ItemID, but to verify that I need the Title of the Page as well.  
In my Sharepoint Site I have 58 pages(Posts). How do I find the Item ID corresponding to each page along with the Title?
